src/audit/template_injection.rs

```diff
@@ -220,12 +177,11 @@ impl TemplateInjection {
                 } else if context.starts_with("matrix.") || context == "matrix" {
                     if let Some(Strategy { matrix, .. }) = &job.strategy {
                         let matrix_is_static = match matrix {
-                            // The matrix is statically defined, but one
-                            // or more keys might contain expressions.
-                            Some(LoE::Literal(matrix)) => self.matrix_is_static(context, matrix),
                             // The matrix is generated by an expression, meaning
                             // that it's trivially not static.
                             Some(LoE::Expr(_)) => false,
+
+                            Some(inner) => models::Matrix::new(inner).is_static(),
```

Comment:
This now became straightforward, but there is an optimization opportunity here by caching the expansion somewhere, so we don't expand the same matrix twice (first when running `self-hosted`, second when running `template-injection`)

Reply:
Yeah, good callout -- one thing we could do here is have `models::Job` contain the expanded `Option<Matrix>` to cache things. But we can think about that as a follow-on 🙂